src/flask_restalchemy/resources/resources.py

```diff
@@ -122,8 +123,13 @@ def get(self, id=None):
                 return NOT_FOUND_ERROR, 404
             return self._serializer.dump(model)
         else:
-            model = query_from_request(self._resource_model, self._serializer, request)
-            return model
+            if self._query_callback:
```

Comment:
You could avoid duplication doing something like this
```suggestion
            query = None
            if self._query_callback:
                query = self._query_callback(self._resource_model)
            query = query_from_request(self._resource_model, self._serializer, request, query=query)
```